osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -327,47 +376,54 @@ private IEnumerable<Memory<PathControlPoint>> convertPoints(ReadOnlyMemory<strin
             // The first control point must have a definite type.
             vertices[0].Type = type;
 
-            // A path can have multiple implicit segments of the same type if there are two sequential control points with the same position.
-            // To handle such cases, this code may return multiple path segments with the final control point in each segment having a non-null type.
-            // For the point string X|1:1|2:2|2:2|3:3, this code returns the segments:
-            // X: { (1,1), (2, 2) }
-            // X: { (3, 3) }
-            // Note: (2, 2) is not returned in the second segments, as it is implicit in the path.
-            int startIndex = 0;
-            int endIndex = 0;
+            return verticesToPoints();
 
-            while (++endIndex < vertices.Length - endPointLength)
+            // Span can't be used in yield generator methods, even if it does not live accross yields
+            // Wrap the yield part into a separated method to workaround this
```

Comment:
```suggestion
            // Span can't be used in yield generator methods, even if it does not live across yields
            // Wrap the yield part into a separated method to workaround this
```

Also this method workaround is... not great in general. But I guess I can look away.

Could this become a proper, separate, pure method? I think I'd hate this less if it could.